tests/test_experiment_server.py

```diff
@@ -256,6 +256,87 @@ def test_records_notification_for_non_mturk_recruiter(self, a, webapp, active_co
         assert models.Notification.query.one().event_type == u'AssignmentSubmitted'
 
 
+@pytest.mark.usefixtures('experiment_dir', 'db_session', 'dummy_mailer')
```

Comment:
Nice tests. I quickly understood (and trusted) what the production code is doing based on these. I don't think you're ever actually _using_ the db_session fixture in any of the tests, so I think the initial `@pytest.mark.usefixtures()` decorator is enough to load it before each test (it can be removed from the method signatures).

Reply:
You're right there was a point in my implementation where I made use of it, but it turned out to be unnecessary, but I didn't remove it. I'll clean that up.

Reply:
Done.